Modules/Office365DSC/DSCResources/MSFT_SPOPropertyBag/MSFT_SPOPropertyBag.psm1

```diff
@@ -0,0 +1,206 @@
+function Get-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $WebUrl,
+
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $Key,
+
+        [Parameter()]
+        [System.String]
+        $Value,
+
+        [Parameter()]
+        [ValidateSet('Present', 'Absent')]
+        [System.String]
+        $Ensure = 'Present',
+
+        [Parameter(Mandatory = $true)]
+        [System.Management.Automation.PSCredential]
+        $GlobalAdminAccount
+    )
+
+    Write-Verbose -Message "Getting configuration of SCCaseHoldPolicy for $Name"
+
+    Test-MSCloudLogin -CloudCredential $GlobalAdminAccount `
+                      -ConnectionUrl $WebUrl `
+                      -Platform PnP
+
+    $property = Get-PnPPropertyBag | Where-Object { $_.Key -eq $Key }
+
+    if ($null -eq $property)
+    {
+        Write-Verbose -Message "SPOPropertyBag $($Key) does not exist at {$Url}."
+        $result = $PSBoundParameters
+        $result.Ensure = 'Absent'
+        return $result
+    }
+    else
+    {
+        Write-Verbose "Found existing SPOPropertyBag Key $Key at {$Url}"
+        $result = @{
+            Ensure             = 'Present'
+            WebUrl             = $WebUrl
+            Key                = $property.Key
+            Value              = $property.Value
+            GlobalAdminAccount = $GlobalAdminAccount
+        }
+
+        Write-Verbose -Message "Get-TargetResource Result: `n $(Convert-O365DscHashtableToString -Hashtable $result)"
+        return $result
+    }
+}
+
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $WebUrl,
+
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $Key,
+
+        [Parameter()]
+        [System.String]
+        $Value,
+
+        [Parameter()]
+        [ValidateSet('Present', 'Absent')]
+        [System.String]
+        $Ensure = 'Present',
+
+        [Parameter(Mandatory = $true)]
+        [System.Management.Automation.PSCredential]
+        $GlobalAdminAccount
+    )
+
+    Write-Verbose -Message "Setting configuration of SPOPropertyBag property for $Key at {$WebUrl}"
+
+    Test-MSCloudLogin -CloudCredential $GlobalAdminAccount `
+                      -ConnectionUrl $WebUrl `
+                      -Platform PnP
+
+    $currentProperty = Get-TargetResource @PSBoundParameters
+
+    if (('Present' -eq $Ensure))
```

Comment:
Remove double brackets

You are not checking the value of currentProperty. What happens if the property already exists? Won't Set-PnPPropertyBagValue throw an error?